shell/platform/android/test/io/flutter/embedding/android/FlutterActivityTest.java

```diff
@@ -196,6 +202,23 @@ public void itCanBeDetachedFromTheEngineAndStopSendingFurtherEvents() {
     verify(mockDelegate, times(1)).onDetach();
   }
 
+  @Test
+  public void itRestoresPluginStateBeforePluginOnCreate() {
+    FlutterLoader mockFlutterLoader = mock(FlutterLoader.class);
+    FlutterJNI mockFlutterJni = mock(FlutterJNI.class);
+    when(mockFlutterJni.isAttached()).thenReturn(true);
+    FlutterEngine cachedEngine =
+        new FlutterEngine(RuntimeEnvironment.application, mockFlutterLoader, mockFlutterJni);
+    FakeFlutterPlugin fakeFlutterPlugin = new FakeFlutterPlugin();
+    cachedEngine.getPlugins().add(fakeFlutterPlugin);
+    FlutterEngineCache.getInstance().put("my_cached_engine", cachedEngine);
+
+    Intent intent =
+        FlutterActivity.withCachedEngine("my_cached_engine").build(RuntimeEnvironment.application);
+    Robolectric.buildActivity(FlutterActivity.class, intent).create();
+    assertTrue("FakeFlutterPlugin onCreate", fakeFlutterPlugin.onCreateCalled);
```

Comment:
Since it's an error message, make it more error-like?

Reply:
Done